### ISSUE TITLE
Explicitly note certificate "purpose" (the subject)

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-libp2p-quic-transport#23]
Currently, if a peer uses their peer ID to sign a certificate, QUIC will allow the signed key to act on behalf of the peer. The peer should have to *explicitly state* that they want to delegate their peer ID.

Motivation: If we don't do this, we won't be able to use peer IDs to sign any certificates *except* delegation certificates.

Proposal: Put `/p2p/QmPeerID` in the certificate's subject field (or some other field if that doesn't make sense). The certificate would only be valid if there exists some certificate higher up the chain signed by that peer ID.

**Comment by marten-seemann** [comment]
It seems like there are two ways we can go from here:
1. If we want to use private keys directly, we can use raw pub key authentication, and we can get rid of the certificate chain (until we actually implement RFC 7250). This means that peers would need to generate child peer IDs if they want to keep their main ID offline.
2. We decide that child peer IDs are not a thing, and that we use certificate chains to enable peers to keep the private key of their main ID offline. This would then require us to fix this issue as well as #21.

**Comment by marten-seemann** [comment]
Yes, RFC allows for a fallback. It's basically just an extension that is sent in the ClientHello TLS message. If the server supports it, it proceeds accordingly. TLS skips unsupported TLS extensions, so we could just use the normal certificate-based authentication then.
Using raw public keys will take us some time anyway. I'm planning to implement it into tls-tris at some point, but I don't see that happening before next year.

My point was that if we decide to use raw public keys (at some point) and to use the certificates as a fallback, this implies that we're rather taking the child peer ID path. Then we don't need any for support certificate chains, a single self-signed certificate would be sufficient.

**Comment by Stebalien** [comment]
I see. Yeah, I'd be fine just going with the single-link chain then. That's what the original TLS patch did anyways.
